### PR TITLE
LPS-193193 Adding tests for autosave in the Data Set pages

### PR DESCRIPTION
**Pull request by JerryAdrianoo**
**Description:**
Adding tests for autosave in the Data Set pages

**JIRA Ticket:**
https://liferay.atlassian.net/browse/LPS-193193

### CODE REVIEW

**Comment by JerryAdrianoo** [comment]
ci:test:frontend

**Comment by JerryAdrianoo** [comment]
ci:stop

**Comment by JerryAdrianoo** [comment]
ci:test:frontend

**Comment by JerryAdrianoo** [comment]
ci:test:sf

**Comment by JerryAdrianoo** [comment]
@CarlosBrichete failures not related 😄

**Comment by JerryAdrianoo** [comment]
I don't know why have a blank space in CanSeeSaveAndCancelButtonOnDetails description. It on appears here in github, but in my code is ok.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewFilters.testcase [hunk 1]
```diff
@@ -1830,6 +1830,81 @@ definition {
 		}
 	}
 
+	@description = "LPS-188333. Confirm that changes in the filters are not autosave"
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewFilters.testcase:1833]: This description is not correct. You are editing and deleting a filter, so you can confirm the changes are autosaved

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewPagination.testcase [hunk 2]
```diff
@@ -298,4 +301,24 @@ definition {
 		}
 	}
 
+	@description = "LPS-188333. Confirm that changes in the pagination are not autosave"
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewPagination.testcase:304]: It should say "autosaved"

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewPagination.testcase [hunk 3]
```diff
@@ -298,4 +301,24 @@ definition {
...
+	@priority = 5
+	test PaginationChangesAreNotAutoSave {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewPagination.testcase:306]: PaginationChangesAreNotAutoSave**d**

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewPagination.testcase [hunk 4]
```diff
@@ -298,4 +301,24 @@ definition {
 		}
 	}
 
+	@description = "LPS-188333. Confirm that changes in the pagination are not autosave"
+	@priority = 5
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewPagination.testcase:305]: Low this to 3

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewSorting.testcase [hunk 5]
```diff
@@ -476,4 +476,60 @@ definition {
 		}
 	}
 
+	@description = "LPS-188333. Confirm that changes in the sorting are not autosave"
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewSorting.testcase:479]: It should say "autosaved"

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewSorting.testcase [hunk 6]
```diff
@@ -476,4 +476,60 @@ definition {
...
+	@priority = 5
+	test SortingChangesAreAutosave {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewSorting.testcase:481]: SortingChangesAreAutosave**d**

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewSorting.testcase [hunk 7]
```diff
@@ -476,4 +476,60 @@ definition {
 		}
 	}
 
+	@description = "LPS-188333. Confirm that changes in the sorting are not autosave"
+	@priority = 5
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewSorting.testcase:480]: Low this to 3

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewSorting.testcase [hunk 8]
```diff
@@ -476,4 +476,60 @@ definition {
...
+
+		task ("When the user reorder the sortings") {
+			DragAndDrop.javaScriptDragAndDropToObject(
+				key_rowPosition = 1,
+				key_text = "Ascending",
+				locator1 = "DataSet#SORT_ENTRY_TABLE",
+				locator2 = "//tr[@class='orderable-table-row'][2]//*[contains(text(), 'id')]");
+		}
+
+		task ("When the user reorder the sortings") {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewSorting.testcase:527]: Repeated task description

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewFilters.testcase [hunk 9]
```diff
@@ -1830,6 +1830,81 @@ definition {
...
+	@priority = 5
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewFilters.testcase:1834]: Low this to 4

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewFilters.testcase [hunk 10]
```diff
@@ -1830,6 +1830,81 @@ definition {
...
+	test FilterChangesAreAutosave {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewFilters.testcase:1835]: FilterChangesAreAutosave**d**

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewFields.testcase [hunk 11]
```diff
@@ -1481,6 +1481,65 @@ definition {
 		}
 	}
 
+	@description = "LPS-188333. Confirm that changes in the fields are not autosave"
+	@priority = 5
+	test FieldsChangesAreAutosave {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewFields.testcase:1486]: FieldsChangesAreAutosave**d**

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewFilters.testcase [hunk 12]
```diff
@@ -1830,6 +1830,81 @@ definition {
...
+		task ("Then the Save and Cancel button are not present") {
```
Comment (CarlosBrichete) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset/DataSetViewFilters.testcase:1836]: Same here

**Review by CarlosBrichete — CHANGES_REQUESTED** [review]
Please check my change requests

**Comment by JerryAdrianoo** [comment]
@CarlosBrichete done!

**Comment by JerryAdrianoo** [comment]
ci:test:sf

**Comment by JerryAdrianoo** [comment]
ci:test:frontend-data-set

**Review by CarlosBrichete — APPROVED** [review]
LGTM

**Comment by JerryAdrianoo** [comment]
ci:forward

**Pull request closed without merging**